fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneOnlineIndexingTest.java

```diff
@@ -644,6 +646,24 @@ protected void openRecordStore(FDBRecordContext context, FDBRecordStoreTestBase.
         setupPlanner(null);
     }
 
+    protected void openRecordStoreWithFilter(FDBRecordContext context, FDBRecordStoreTestBase.RecordMetaDataHook hook, boolean filterOut) {
+        RecordMetaDataBuilder metaDataBuilder = RecordMetaData.newBuilder().setRecords(TestRecordsTextProto.getDescriptor());
+        metaDataBuilder.getRecordType(COMPLEX_DOC).setPrimaryKey(concatenateFields("group", "doc_id"));
+        hook.apply(metaDataBuilder);
+        final FDBRecordStore.Builder builder = getStoreBuilder(context, metaDataBuilder.getRecordMetaData())
+                .setSerializer(TextIndexTestUtils.COMPRESSING_SERIALIZER);
+        if (filterOut) {
+            recordStore = builder
+                    .setIndexMaintenanceFilter((i, r) -> IndexMaintenanceFilter.IndexValues.NONE)
```

Comment:
It seems worthwhile to test predicate support too, perhaps parameterizing this method, and the eventual test for whether to filter with a predicate or a filter.

Reply:
√